lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AddressSessionService.java

```diff
@@ -294,6 +289,67 @@ private boolean validSignature(SignedJWT signedJWT, PublicKey clientPublicKey)
         }
     }
 
+    public List<CanonicalAddressWithResidency> parseAddresses(String addressBody)
+            throws AddressProcessingException {
+        List<CanonicalAddressWithResidency> addresses;
```

Comment:
I'd like to see a test that checks that we can marshall a String to a List<CanonicalAddressWithResidency>. We could test this with integration tests but I think there's value in checking jackson is doing its job. We might want to check how the Optional<> setters work